hs/src/Reach/Eval/Core.hs

```diff
@@ -3275,6 +3323,11 @@ evalStmt = \case
   (s@(JSBreak a _ _) : _) -> illegal a s "break"
   (s@(JSLet a _ _) : _) -> illegal a s "let"
   (s@(JSClass a _ _ _ _ _ _) : _) -> illegal a s "class"
+  (JSExpressionStatement (JSStringLiteral a hs) sp) : ks
+    | trimQuotes hs == "use strict" -> do
+      use_strict <- asks $ sco_use_strict . e_sco
+      liftIO $ writeIORef use_strict True
```

Comment:
Hmm... this seems dangerous. From your docs, I assumed `use_strict` would be something that you would `local (\e -> e { sco = old sco { sco_use_strict = True } }` are the `evalStmt ks`. By making it a mutation... doesn't this mean that something like:

```
const f = (x, y) => y;
'use strict';
f(1, 2);
```

will error, because the body of the closure doesn't get evaluated until after the modification has happened?

Reply:
That code segment will not error because the closure was declared in non-strict mode. When we create the `SLV_Clo` and convert sco to cloenv, we will copy what `use_strict` was set to at the time of declaration and use that when evaluating `SLV_Clo`.

Reply:
How about this then:

```
switch (x) {
 case Var1: {
  'use strict';
  const y1 = 1;
  return y1 + 1; }
 case Var2: {
  const y2 = 2;
  return 2; }
}
```

I believe that `y2` will be protected when it should not, because each case shares the same scope

Reply:
That example compiles fine, the strict mode currently follows "syntactic scope" if that makes sense as opposed to how our evaluation which might use the same env for different blocks/continuations. It would treat the two `case` blocks as adjacent scopes as opposed to one scope

Reply:
I'm scared Chris. Mutation is scary. Does it not work if it is `local`?